src/assets/scripts/client/trafficGenerator/SpawnPatternModel.js

```diff
@@ -0,0 +1,487 @@
+import _random from 'lodash/random';
+import BaseModel from '../base/BaseModel';
+import RouteModel from '../airport/Route/RouteModel';
+import { bearingToPoint } from '../math/flightMath';
+import { AIRPORT_CONSTANTS } from '../constants/airportConstants';
+import { FLIGHT_CATEGORY } from '../constants/aircraftConstants';
+import { TIME } from '../constants/globalConstants';
+
+/**
+ * Defines a spawn pattern for a specific route within the area
+ *
+ * This can be for departures or arrivals. Provides a starting point and definition
+ * for all spawning aircraft
+ *
+ * @class SpawnPatternModel
+ * @extends BaseModel
+ */
+export default class SpawnPatternModel extends BaseModel {
+    /**
+     * @constructor
+     * @for SpawnPatternModel
+     * @param spawnPatternJson {object}
+     * @param navigationLibrary {NavigationLibrary}
+     */
+    /* istanbul ignore next */
+    constructor(spawnPatternJson, navigationLibrary) {
+        super(spawnPatternJson, navigationLibrary);
+
+        if (!_isObject(spawnPatternJson) || _isEmpty(spawnPatternJson)) {
+            throw new TypeError('Invalid parameter passed to SpawnPatternModel');
+        }
+
+        if (!_isObject(navigationLibrary) || _isEmpty(navigationLibrary)) {
+            throw new TypeError('Invalid NavigationLibrary passed to SpawnPatternModel');
+        }
+
+        /**
+         * Interval id
+         *
+         * Stored here so a specific interval can be associated with a
+         * specfic `SpawnPatternModel` instance. An Interval may be reset
+         * or changed during the life of the app.
+         *
+         * Provides easy access to a specif interval id
+         *
+         * @property scheduleId
+         * @type {number}
+         * @default -1
+         */
+        this.scheduleId = -1;
+
+        /**
+         * One of `FLIGHT_CATEGORY`
+         *
+         * @property category
+         * @type {string}
+         * @default ''
+         */
+        this.category = '';
+
+        /**
+         * Type of arrival or departure pattern
+         *
+         * Could be `random`, `cyclic`. `surge` or `wave`
+         *
+         * @property type
+         * @type {string}
+         * @default ''
+         */
+        this.method = '';
+
+        /**
+         *
+         *
+         * @property origin
+         * @type {string}
+         * @default ''
+         */
+        this.origin = '';
+
+        /**
+         *
+         *
+         * @property destination
+         * @type {string}
+         * @default ''
+         */
+        this.destination = '';
+
+        /**
+         * String representation of a `StandardRoute`
+         *
+         * @property route
+         * @type {string}
+         * @default
+         */
+        this.route = '';
+
+        /**
+         * Rate at which aircaft spawn, express in aircraft per hour
+         *
+         * @property rate
+         * @type {number}
+         * @default -1
+         */
+        this.rate = -1;
```

Comment:
the thought there is that if it does explode the implementor didn't write the `spawnPattern` correctly.  within the `.init()` method we do: 
```javascript
this.rate = spawnPatternJson.rate;
```
so as long as the author defines a `spawnPattern` correctly, we should be golden.